src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToCustomizeTheApphost.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.IO;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToCustomizeTheApphost : SdkTest
+    {
+        public GivenThatWeWantToCustomizeTheApphost(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [WindowsOnlyFact]
+        public void It_contains_resources_from_the_managed_dll()
```

Comment:
I think we should test both file version and icon in some way - since those are our primary scenarios.
Maybe even the manifest...
I know that the test are essentially doing the same thing - but it's covering the end-to-end behaviors we want from this.

Reply:
I agree. It would be nice to have tests that demonstrate the end-to-end usage of all of these.

---
In reply to: [219612834](https://github.com/dotnet/sdk/pull/2545#discussion_r219612834) [](ancestors = 219612834)

Reply:
I don't think this is worth doing due to the amount of work it would take to check that the icon or manifest is correct in the output - unless there's an existing end-to-end test for resources in roslyn, but I couldn't find anything like that.

Reply:
If it's too costly, I can live with not testing this.

Reply:
Same here - if you tried and it looks like a lot of work - manual testing it is for now.